packages/payment-processor/src/payment/encoder-payment.ts

```diff
@@ -10,25 +10,27 @@ import { prepareSwapToPayAnyToErc20Request } from './swap-any-to-erc20';
 import { prepareEthProxyPaymentTransaction } from './eth-proxy';
 import { prepareEthFeeProxyPaymentTransaction } from './eth-fee-proxy';
 import { prepareAnyToEthProxyPaymentTransaction } from './any-to-eth-proxy';
+import { prepareErc777StreamPaymentTransaction } from './erc777-stream';
 import { IConversionPaymentSettings } from '.';
 import { getPaymentNetworkExtension } from './utils';
 
-export function encodeRequestPayment(
+export async function encodeRequestPayment(
   request: ClientTypes.IRequestData,
   provider: providers.Provider,
   options?: IRequestPaymentOptions,
-): IPreparedTransaction {
+): Promise<IPreparedTransaction> {
   if (options && options.swap) {
     return encodeRequestPaymentWithSwap(request, provider, options);
   } else {
-    return encodeRequestPaymentWithoutSwap(request, options);
+    return await encodeRequestPaymentWithoutSwap(request, provider, options);
   }
 }
 
-export function encodeRequestPaymentWithoutSwap(
+export async function encodeRequestPaymentWithoutSwap(
   request: ClientTypes.IRequestData,
+  provider: providers.Provider,
   options?: IRequestPaymentOptions,
-): IPreparedTransaction {
+): Promise<IPreparedTransaction> {
```

Comment:
NOK for this breaking change. I understand where it comes from but I think it's not worth it. You should instead, as a first step at least, add an equivalent of `encodeRequestPaymentWithoutSwap`, async and specific to ERC777, and handle this case separately in your back-end.

Reply:
OK I'm preparing it to handle it separately in the API backend.

Reply:
Replaced by #896